crates/ra_editor/src/typing.rs

```diff
@@ -136,6 +136,58 @@ pub fn on_eq_typed(file: &SourceFileNode, offset: TextUnit) -> Option<LocalEdit>
     })
 }
 
+pub fn on_dot_typed(file: &SourceFileNode, offset: TextUnit) -> Option<LocalEdit> {
+    let before_dot_offset = offset - TextUnit::of_char('.');
+
+    let _whitespace = find_leaf_at_offset(file.syntax(), before_dot_offset).left_biased()?;
+
+    // find whitespace just left of the dot
+    ast::Whitespace::cast(_whitespace)?;
+
+    // make sure there is a method call
+    let _method_call = _whitespace
+        .siblings(Direction::Prev)
+        // first is whitespace
+        .skip(1)
+        .next()?;
+
+    ast::MethodCallExprNode::cast(_method_call)?;
+
+    // find how much the _method call is indented
+    let method_chain_indent = _method_call
+        .ancestors()
+        .skip(1)
+        .next()?
```

Comment:
`.acenstors.skip(1).next()` can be written more succinctly as `.parent()` I think?